Lib/pathlib.py

```diff
@@ -198,14 +213,34 @@ def resolve(self, path, strict=False):
         # Means fallback on absolute
         return None
 
-    def _split_extended_path(self, s, ext_prefix=ext_namespace_prefix):
+    def _split_extended_path(self, s, ext_prefix=ext_namespace_prefix,\
+                             ext_prefix2=ext_namespace_prefix2):
+        # See https://bugs.python.org/issue33898
         prefix = ''
-        if s.startswith(ext_prefix):
+        if s.startswith(ext_prefix) or s.startswith(ext_prefix2):
             prefix = s[:4]
             s = s[4:]
-            if s.startswith('UNC\\'):
-                prefix += s[:3]
-                s = '\\' + s[3:]
+            index = s.find('\\')
+            if index != -1:
+                # Do not assume case sensitivity.
+                # See https://docs.microsoft.com/ru-ru/windows/desktop/FileIO/naming-a-file
+                # But .lower() is not recommended. See https://bugs.python.org/issue32612
+                s1 = s[:index]
+                if s1 == 'Global' or s1 == 'GLOBAL':
+                    prefix += s[:6]
+                # For example, Path('//?/Global/Z:/').drive
+                    if s[8] == ':':
```

Comment:
What should we do if the path is `//?/Global/Z`? Looks like we'll get an IndexError here in that case.

Reply:
Fixed. Thanks.